crates/sui-core/src/authority_server.rs

```diff
@@ -1033,6 +1035,23 @@ impl ValidatorService {
             }
             .into()
         );
+
+        // We set the soft bundle max size to be half of the consensus max transactions in block size. We do this to account for
+        // serialisation overheads and to ensure that the soft bundle is not too large when is attempted to be posted via consensus.
+        // Although half the block size is on the extreme side, it's should be good enough for now.
+        let soft_bundle_max_size_bytes =
+            protocol_config.consensus_max_transactions_in_block_bytes() / 2;
```

Comment:
I know that this is not ideal, but until we have a way from consensus submission to propagate the error all the way up to the API, we should probably be on the safe side here. The serialised transaction size is not the same as the one that is finally submitted to consensus (the later is always bigger) as it's wrapper on additional structures.

Alternatively, we could do those checks at the point when the consensus transactions are getting created https://github.com/MystenLabs/sui/blob/22c1ef1d698b013bfb06f2195869c0333471eb32/crates/sui-core/src/authority_server.rs#L723 (serialise them and check the size). I am not sure that's better approach - for now I would be in favour of keeping this simpler.